scripts/cli/src/util/checkBridgeBalance.ts

```diff
@@ -0,0 +1,108 @@
+import { ApiSingleton } from "../util/init";
+import PrettyError from "pretty-error";
+
+async function main(): Promise<void> {
+	const api = await ApiSingleton.getInstance();
+	const BRIDGE_LIMIT = 500_000_000;
+	const BRIDGE_POLY_TOTAL = 1_000_000_000;
+	const BRIDGE_TIME_PERIOD = hoursToSeconds(2); //e.g 4 * 3600
+	let count = 0;
+	let userArray: {
+		recipient: string;
+		polyx: number;
+		lastTimeBridged: number;
+	}[] = [];
+
+	function currentTimeToSeconds() {
+		const currentDateTime = new Date();
+		const resultInSeconds = currentDateTime.getTime() / 1000;
+		return resultInSeconds;
+	}
+
+	function hoursToSeconds(hours: number) {
+		let seconds = hours * 3600;
+		return seconds;
+	}
+
+	// Subscribe to the new headers on-chain. The callback is fired when new headers
+	// are found, the call itself returns a promise with a subscription that can be
+	// used to unsubscribe from the newHead subscription
+	const unsubscribe = await api.rpc.chain.subscribeNewHeads((header) => {
+		console.log(`Chain is at block: #${header.number}`);
+
+		// Subscribe to system events via storage
+		api.query.system.events((events) => {
+			console.log(`\nReceived ${events.length} events:`);
+
+			// Loop through the Vec<EventRecord>
+			events.forEach((record) => {
+				// Extract the phase, event and the event types
+				const { event, phase } = record;
+				const types = event.typeDef;
+				if (event.section === "bridge") {
+					if (event.method === "BridgeTxScheduled") {
+						// Loop through each of the parameters, displaying the type and data
+						event.data.forEach((data, index) => {
+							if (types[index].type === "BridgeTx") {
+								const txData = JSON.parse(data.toString());
+
+								if (
+									userArray.length === 0 ||
+									!userArray.some((user) => user.recipient === txData.recipient)
+								) {
```

Comment:
The code still uses an array for storing all bridge transactions.
```
// key = recipient, value = array of bridge transaction for the `recipient`
let users = new Map();

/// process a bridge transaction
let userTxData = users.get(txData.recipient);
if (userTxData) {
  // Already has transactions for this user.  (`checksUser` will need to return an updated array for the user).
  userTxData = checksUser(userTxData, txData);
} else {
  // New user, create new array of transaction for this user.
  userTxData = [{
	recipient: txData.recipient,
	polyx: txData.value,
	lastTimeBridged: currentTimeToSeconds(),
  }];
}
// update stored user data.
users.put(txData.recipient, userTxData);
```

Reply:
Ahh, makes sense I had the wrong map in mind, at first I thought you were referring to array.map instead of new Map(); would make the relevant changes. thanks for the advice.

Reply:
I think keeping the per-user txdata separate would also make it easier to expire/remove old transactions (older then `BRIDGE_TIME_PERIOD`) and sum-up the non-expired transactions to get the per-user total balance transferred.

Also I just noticed that the current script doesn't see to remove old transactions, so the memory usage will just keep growing.